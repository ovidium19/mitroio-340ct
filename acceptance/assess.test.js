import puppeteer from 'puppeteer'

describe('testUser takes assessment', () => {
    let browser
    let page

    beforeAll(() => {
        jest.setTimeout(99000)
    })

    test('User logs in, selects Test Course, proceeds to assessment, takes assessment, passess with 100%, match snapshot', async done => {
        let elem
        browser = await puppeteer.launch()
        page = await browser.newPage()
        await page.goto('http://localhost:8000')
        elem = await page.waitFor('[href="/account/login"]')
        await elem.click()
        await page.type('#usernameid','testUser')
        await page.type('#passwordid','testPassword')
        elem = await page.$('[type="submit"]')
        await elem.click()
        await page.waitFor('#testcourse')
        elem = await page.waitFor('[href="/courses"]')
        await elem.click()
        elem = await page.waitFor('#testcourse button')
        await elem.click()
        elem = await page.waitFor('[href^="/course/"]')
        await elem.click()
        elem = await page.waitFor('.page-content button')
        let text = await page.$eval('.page-content button', (e) => e.innerHTML)
        expect(text).toEqual('Go to Assessment')
        await elem.click()
        elem = await page.waitFor('.start-assess button')
        text = await page.$eval('.start-assess button', (e) => e.innerHTML)
        expect(text).toEqual('Start Assessment')
        await elem.click()
        elem = await page.waitFor('#answer4')
        await elem.click()
        let className = await page.$eval('#answer4', e => e.className)
        expect(/active/.test(className)).toBe(true)
        await elem.click()
        elem = await page.waitFor('.question-answer-submit button')
        await elem.click()
        expect(page.url()).toEqual('http://localhost:8000/course/5bf5b9d51c9d440000c6f2c7/assess')
        elem = await page.waitFor('.score')
        text = await page.$eval('.score', e => e.innerHTML)
        expect(text).toEqual('100%')
        let content = await page.content()
        expect(content).toMatchSnapshot()
        await browser.close()

        done()
    })
})
