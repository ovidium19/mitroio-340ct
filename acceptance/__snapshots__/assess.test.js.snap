// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testUser takes assessment User logs in, selects Test Course, proceeds to assessment, takes assessment, passess with 100%, match snapshot 1`] = `
"<!DOCTYPE html><html lang=\\"en\\"><head>
    <meta charset=\\"UTF-8\\">
    <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1.0\\">
    <meta http-equiv=\\"X-UA-Compatible\\" content=\\"ie=edge\\">
    <link rel=\\"stylesheet\\" href=\\"https://stackpath.bootstrapcdn.com/bootstrap/4.1.3/css/bootstrap.min.css\\" integrity=\\"sha384-MCw98/SFnGE8fJT3GXwEOngsV7Zt27NXFoaoApmYm81iuXoPkFOJwJ8ERdknLPMO\\" crossorigin=\\"anonymous\\">
    <link href=\\"https://fonts.googleapis.com/icon?family=Material+Icons\\" rel=\\"stylesheet\\">
    <link rel=\\"stylesheet\\" href=\\"https://use.fontawesome.com/releases/v5.5.0/css/all.css\\" integrity=\\"sha384-B4dIYHKNBt8Bc12p+WXckhzcICo0wtJAoU8YZTY5qE0Id1GSseTk6S+L3BlXeVIU\\" crossorigin=\\"anonymous\\">
    <script src=\\"https://code.jquery.com/jquery-3.3.1.slim.min.js\\" integrity=\\"sha384-q8i/X+965DzO0rT7abK41JStQIAqVgRVzpbzo5smXKp4YfRvH+8abtTE1Pi6jizo\\" crossorigin=\\"anonymous\\"></script>
    <script src=\\"https://cdnjs.cloudflare.com/ajax/libs/popper.js/1.14.3/umd/popper.min.js\\" integrity=\\"sha384-ZMP7rVo3mIykV+2+9J3UJ46jBk0WLaUAdn689aCwoqbBJiSnjAK/l8WvCWPIPm49\\" crossorigin=\\"anonymous\\"></script>
    <script src=\\"https://stackpath.bootstrapcdn.com/bootstrap/4.1.3/js/bootstrap.min.js\\" integrity=\\"sha384-ChfqqxuZUCnJSK3+MXmPNIyE6ZbWh2IMqE241rYiqJxyMiZ6OW/JmZQ5stwEULTy\\" crossorigin=\\"anonymous\\"></script>
    <title>Document</title>
<style type=\\"text/css\\">.navbar-text {
  line-height: 2;
}
</style><style type=\\"text/css\\">.form-container .form-box-centered {
  box-shadow: 1px 1px 1px 1px black;
  padding: 1.3em;
}
</style><style type=\\"text/css\\">.courses .course-list {
  background-color: lightgray;
  justify-content: space-around;
  align-content: stretch;
}
.courses .course-list .course-item {
  min-height: 250px;
}
.courses .course-list .course-item .card-header {
  font-weight: bold;
  border-bottom: 1px solid white;
}
.courses .course-list .course-item .course-description {
  min-height: 80px;
}
.courses .course-list .course-item .course-category {
  color: greenyellow;
}
@media all and (max-width: 1149px) {
  .courses .course-list .course-item {
    margin: 1em 0;
    flex: 0 0 50%;
  }
}
@media all and (max-width: 680px) {
  .courses .course-list .course-item {
    min-height: 200px;
    margin: 1em 10%;
    flex: 0 0 80%;
  }
}
@media all and (min-width: 1150px) {
  .courses .course-list .course-item {
    flex: 0 0 33%;
  }
}
.course-landing-content {
  align-content: stretch;
  flex: 1;
}
.course-landing-content .course-page-list {
  padding: 0;
  margin: 0;
}
.course-landing-content .course-page-list .course-menu-item {
  padding: 0.4em;
  border-bottom: 1px solid white;
}
.course-landing-content .course-page-list .course-menu-item p {
  vertical-align: middle;
  line-height: 1.5;
}
.course-landing-content .course-page-list .course-menu-item .go-to-page {
  border: none;
  background: none;
  font-size: 1.3em;
}
.course-landing-content .course-page-list .course-menu-item .go-to-page[disabled] {
  color: orange;
}
.course-landing-content .course-page-list .course-menu-item .go-to-page:not([disabled]) {
  color: greenyellow;
}
</style><style type=\\"text/css\\">.page-content div.bash-block {
  overflow-wrap: normal;
}
.page-content div.bash-block pre.bash {
  color: white;
  font-size: medium ;
  font-family: Consolas,Monaco,Lucida Console,Liberation Mono,DejaVu Sans Mono,Bitstream Vera Sans Mono,Courier New, monospace;
  display: inline-block;
}
.start-assess {
  flex-grow: 2;
}
ul.page-list {
  list-style-type: circle;
}
</style><style type=\\"text/css\\">.question-box {
  box-shadow: 1px 1px 1px 1px lightgray inset;
}
.question-box .card-header {
  font-size: 1.3em;
  font-family: 'Lucida Sans', 'Lucida Sans Regular', 'Lucida Grande', 'Lucida Sans Unicode', Geneva, Verdana, sans-serif;
  background: bisque;
  box-shadow: 1px 1px 3px 1px lightgray inset;
  color: black;
}
.question-answer {
  font-size: 1.2em;
  font-family: 'Lucida Sans', 'Lucida Sans Regular', 'Lucida Grande', 'Lucida Sans Unicode', Geneva, Verdana, sans-serif;
  border: 1px solid lightgrey;
  cursor: pointer;
}
.question-answer:hover {
  background: lightcyan;
}
.question-answer .question-answer-text {
  padding-top: 0.3em;
}
.question-answer > .question-answer-text.active {
  background: orangered;
}
.question-answer .question-answer-text.active {
  color: white;
}
</style><style type=\\"text/css\\">.toast-title{font-weight:700}.toast-message{-ms-word-wrap:break-word;word-wrap:break-word}.toast-message a,.toast-message label{color:#FFF}.toast-message a:hover{color:#CCC;text-decoration:none}.toast-close-button{position:relative;right:-.3em;top:-.3em;float:right;font-size:20px;font-weight:700;color:#FFF;-webkit-text-shadow:0 1px 0 #fff;text-shadow:0 1px 0 #fff;opacity:.8;-ms-filter:progid:DXImageTransform.Microsoft.Alpha(Opacity=80);filter:alpha(opacity=80);line-height:1}.toast-close-button:focus,.toast-close-button:hover{color:#000;text-decoration:none;cursor:pointer;opacity:.4;-ms-filter:progid:DXImageTransform.Microsoft.Alpha(Opacity=40);filter:alpha(opacity=40)}.rtl .toast-close-button{left:-.3em;float:left;right:.3em}button.toast-close-button{padding:0;cursor:pointer;background:0 0;border:0;-webkit-appearance:none}.toast-top-center{top:0;right:0;width:100%}.toast-bottom-center{bottom:0;right:0;width:100%}.toast-top-full-width{top:0;right:0;width:100%}.toast-bottom-full-width{bottom:0;right:0;width:100%}.toast-top-left{top:12px;left:12px}.toast-top-right{top:12px;right:12px}.toast-bottom-right{right:12px;bottom:12px}.toast-bottom-left{bottom:12px;left:12px}#toast-container{position:fixed;z-index:999999;pointer-events:none}#toast-container *{-moz-box-sizing:border-box;-webkit-box-sizing:border-box;box-sizing:border-box}#toast-container>div{position:relative;pointer-events:auto;overflow:hidden;margin:0 0 6px;padding:15px 15px 15px 50px;width:300px;-moz-border-radius:3px;-webkit-border-radius:3px;border-radius:3px;background-position:15px center;background-repeat:no-repeat;-moz-box-shadow:0 0 12px #999;-webkit-box-shadow:0 0 12px #999;box-shadow:0 0 12px #999;color:#FFF;opacity:.8;-ms-filter:progid:DXImageTransform.Microsoft.Alpha(Opacity=80);filter:alpha(opacity=80)}#toast-container>div.rtl{direction:rtl;padding:15px 50px 15px 15px;background-position:right 15px center}#toast-container>div:hover{-moz-box-shadow:0 0 12px #000;-webkit-box-shadow:0 0 12px #000;box-shadow:0 0 12px #000;opacity:1;-ms-filter:progid:DXImageTransform.Microsoft.Alpha(Opacity=100);filter:alpha(opacity=100);cursor:pointer}#toast-container>.toast-info{background-image:url(data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABgAAAAYCAYAAADgdz34AAAAAXNSR0IArs4c6QAAAARnQU1BAACxjwv8YQUAAAAJcEhZcwAADsMAAA7DAcdvqGQAAAGwSURBVEhLtZa9SgNBEMc9sUxxRcoUKSzSWIhXpFMhhYWFhaBg4yPYiWCXZxBLERsLRS3EQkEfwCKdjWJAwSKCgoKCcudv4O5YLrt7EzgXhiU3/4+b2ckmwVjJSpKkQ6wAi4gwhT+z3wRBcEz0yjSseUTrcRyfsHsXmD0AmbHOC9Ii8VImnuXBPglHpQ5wwSVM7sNnTG7Za4JwDdCjxyAiH3nyA2mtaTJufiDZ5dCaqlItILh1NHatfN5skvjx9Z38m69CgzuXmZgVrPIGE763Jx9qKsRozWYw6xOHdER+nn2KkO+Bb+UV5CBN6WC6QtBgbRVozrahAbmm6HtUsgtPC19tFdxXZYBOfkbmFJ1VaHA1VAHjd0pp70oTZzvR+EVrx2Ygfdsq6eu55BHYR8hlcki+n+kERUFG8BrA0BwjeAv2M8WLQBtcy+SD6fNsmnB3AlBLrgTtVW1c2QN4bVWLATaIS60J2Du5y1TiJgjSBvFVZgTmwCU+dAZFoPxGEEs8nyHC9Bwe2GvEJv2WXZb0vjdyFT4Cxk3e/kIqlOGoVLwwPevpYHT+00T+hWwXDf4AJAOUqWcDhbwAAAAASUVORK5CYII=)!important}#toast-container>.toast-error{background-image:url(data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABgAAAAYCAYAAADgdz34AAAAAXNSR0IArs4c6QAAAARnQU1BAACxjwv8YQUAAAAJcEhZcwAADsMAAA7DAcdvqGQAAAHOSURBVEhLrZa/SgNBEMZzh0WKCClSCKaIYOED+AAKeQQLG8HWztLCImBrYadgIdY+gIKNYkBFSwu7CAoqCgkkoGBI/E28PdbLZmeDLgzZzcx83/zZ2SSXC1j9fr+I1Hq93g2yxH4iwM1vkoBWAdxCmpzTxfkN2RcyZNaHFIkSo10+8kgxkXIURV5HGxTmFuc75B2RfQkpxHG8aAgaAFa0tAHqYFfQ7Iwe2yhODk8+J4C7yAoRTWI3w/4klGRgR4lO7Rpn9+gvMyWp+uxFh8+H+ARlgN1nJuJuQAYvNkEnwGFck18Er4q3egEc/oO+mhLdKgRyhdNFiacC0rlOCbhNVz4H9FnAYgDBvU3QIioZlJFLJtsoHYRDfiZoUyIxqCtRpVlANq0EU4dApjrtgezPFad5S19Wgjkc0hNVnuF4HjVA6C7QrSIbylB+oZe3aHgBsqlNqKYH48jXyJKMuAbiyVJ8KzaB3eRc0pg9VwQ4niFryI68qiOi3AbjwdsfnAtk0bCjTLJKr6mrD9g8iq/S/B81hguOMlQTnVyG40wAcjnmgsCNESDrjme7wfftP4P7SP4N3CJZdvzoNyGq2c/HWOXJGsvVg+RA/k2MC/wN6I2YA2Pt8GkAAAAASUVORK5CYII=)!important}#toast-container>.toast-success{background-image:url(data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABgAAAAYCAYAAADgdz34AAAAAXNSR0IArs4c6QAAAARnQU1BAACxjwv8YQUAAAAJcEhZcwAADsMAAA7DAcdvqGQAAADsSURBVEhLY2AYBfQMgf///3P8+/evAIgvA/FsIF+BavYDDWMBGroaSMMBiE8VC7AZDrIFaMFnii3AZTjUgsUUWUDA8OdAH6iQbQEhw4HyGsPEcKBXBIC4ARhex4G4BsjmweU1soIFaGg/WtoFZRIZdEvIMhxkCCjXIVsATV6gFGACs4Rsw0EGgIIH3QJYJgHSARQZDrWAB+jawzgs+Q2UO49D7jnRSRGoEFRILcdmEMWGI0cm0JJ2QpYA1RDvcmzJEWhABhD/pqrL0S0CWuABKgnRki9lLseS7g2AlqwHWQSKH4oKLrILpRGhEQCw2LiRUIa4lwAAAABJRU5ErkJggg==)!important}#toast-container>.toast-warning{background-image:url(data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABgAAAAYCAYAAADgdz34AAAAAXNSR0IArs4c6QAAAARnQU1BAACxjwv8YQUAAAAJcEhZcwAADsMAAA7DAcdvqGQAAAGYSURBVEhL5ZSvTsNQFMbXZGICMYGYmJhAQIJAICYQPAACiSDB8AiICQQJT4CqQEwgJvYASAQCiZiYmJhAIBATCARJy+9rTsldd8sKu1M0+dLb057v6/lbq/2rK0mS/TRNj9cWNAKPYIJII7gIxCcQ51cvqID+GIEX8ASG4B1bK5gIZFeQfoJdEXOfgX4QAQg7kH2A65yQ87lyxb27sggkAzAuFhbbg1K2kgCkB1bVwyIR9m2L7PRPIhDUIXgGtyKw575yz3lTNs6X4JXnjV+LKM/m3MydnTbtOKIjtz6VhCBq4vSm3ncdrD2lk0VgUXSVKjVDJXJzijW1RQdsU7F77He8u68koNZTz8Oz5yGa6J3H3lZ0xYgXBK2QymlWWA+RWnYhskLBv2vmE+hBMCtbA7KX5drWyRT/2JsqZ2IvfB9Y4bWDNMFbJRFmC9E74SoS0CqulwjkC0+5bpcV1CZ8NMej4pjy0U+doDQsGyo1hzVJttIjhQ7GnBtRFN1UarUlH8F3xict+HY07rEzoUGPlWcjRFRr4/gChZgc3ZL2d8oAAAAASUVORK5CYII=)!important}#toast-container.toast-bottom-center>div,#toast-container.toast-top-center>div{width:300px;margin-left:auto;margin-right:auto}#toast-container.toast-bottom-full-width>div,#toast-container.toast-top-full-width>div{width:96%;margin-left:auto;margin-right:auto}.toast{background-color:#030303}.toast-success{background-color:#51A351}.toast-error{background-color:#BD362F}.toast-info{background-color:#2F96B4}.toast-warning{background-color:#F89406}.toast-progress{position:absolute;left:0;bottom:0;height:4px;background-color:#000;opacity:.4;-ms-filter:progid:DXImageTransform.Microsoft.Alpha(Opacity=40);filter:alpha(opacity=40)}@media all and (max-width:240px){#toast-container>div{padding:8px 8px 8px 50px;width:11em}#toast-container>div.rtl{padding:8px 50px 8px 8px}#toast-container .toast-close-button{right:-.2em;top:-.2em}#toast-container .rtl .toast-close-button{left:-.2em;right:.2em}}@media all and (min-width:241px) and (max-width:480px){#toast-container>div{padding:8px 8px 8px 50px;width:18em}#toast-container>div.rtl{padding:8px 50px 8px 8px}#toast-container .toast-close-button{right:-.2em;top:-.2em}#toast-container .rtl .toast-close-button{left:-.2em;right:.2em}}@media all and (min-width:481px) and (max-width:768px){#toast-container>div{padding:15px 15px 15px 50px;width:25em}#toast-container>div.rtl{padding:15px 50px 15px 15px}}</style><style type=\\"text/css\\">html,
body {
  height: 100%;
  margin: 0;
}
#app {
  min-height: 100%;
}
</style></head>
<body>
    <div id=\\"app\\" class=\\"container-fluid px-0 d-flex flex-column\\"><nav class=\\"navbar navbar-expand-sm navbar-dark bg-dark\\"><a class=\\"navbar-brand\\" href=\\"/\\"><i class=\\"fa fa-user-graduate fa-3x text-white\\"></i> </a><button class=\\"navbar-toggler\\" type=\\"button\\" data-toggle=\\"collapse\\" data-target=\\"#navLinks\\"><span class=\\"navbar-toggler-icon\\"></span></button><div class=\\"collapse navbar-collapse\\" id=\\"navLinks\\"><ul class=\\"navbar-nav\\"><li class=\\"nav-item \\"><a class=\\"nav-link active\\" aria-current=\\"page\\" href=\\"/\\">Home</a></li><li class=\\"nav-item \\"><a class=\\"nav-link\\" href=\\"/courses\\">Learn</a></li><li class=\\"nav-item \\"><a class=\\"nav-link\\" href=\\"/user/testUser/hub\\">Your Hub</a></li><li class=\\"nav-item \\"><a class=\\"nav-link\\" href=\\"/about\\">About</a></li></ul><ul class=\\"navbar-nav ml-auto\\"><li class=\\"nav-item\\"><a class=\\"nav-link\\" href=\\"/user/testUser/account\\">Account</a></li><li class=\\"nav-item\\"><a class=\\"nav-link\\" href=\\"/user/testUser/signout\\">Sign Out</a></li><span class=\\"navbar-text text-light small\\">Welcome testUser</span></ul></div></nav><div class=\\"container-fluid course-landing-content row px-0 mx-0\\"><div class=\\"p-3 m-0 w-100\\"><p class=\\"lead mb-3 text-center\\">testcourse - Assessment</p><div class=\\"text-center\\"><div class=\\"text-center lead my-3\\"><span>Congratulations</span></div><div class=\\"text-center\\"><span>You have successfully passed the assessment with </span><p class=\\"text-center display-4 text-success score\\">100%</p></div><div class=\\"mt-3 d-flex justify-content-around\\"><a class=\\"btn btn-success\\" href=\\"/user/testUser/hub\\">Your Hub</a><a class=\\"btn btn-success\\" href=\\"/course/5bf5b9d51c9d440000c6f2c7\\">See Course Content</a></div></div></div></div></div>

    <script src=\\"/bundle.js\\"></script>


</body></html>"
`;
